src/corehost/cli/hostpolicy.cpp

```diff
@@ -1,279 +1,376 @@
+
+            pal::string_t app_context_deps_str;
+            fx_definition_vector_t::iterator fx_curr = fx_begin;
+            while (fx_curr != fx_end)
+            {
+                if (fx_curr != fx_begin)
+                    app_context_deps_str += _X(';');
 
-    bool set_app_paths = false;
+                app_context_deps_str += (*fx_curr)->get_deps_file();
+                ++fx_curr;
+            }
 
-    // Runtime options config properties.
-    for (int i = 0; i < g_init.cfg_keys.size(); ++i)
-    {
-        // Provide opt-in compatible behavior by using the switch to set APP_PATHS
-        if (pal::cstrcasecmp(g_init.cfg_keys[i].data(), "Microsoft.NETCore.DotNetHostPolicy.SetAppPaths") == 0)
+            pal::pal_clrstring(app_context_deps_str, &_app_context_deps);
+            pal::pal_clrstring(_resolver.get_lookup_probe_directories(), &_probe_directories);
+
+            if (_resolver.is_framework_dependent())
+            {
+                pal::pal_clrstring(get_root_framework(fx_definitions).get_found_version(), &_clr_library_version);
+            }
+            else
+            {
+                pal::pal_clrstring(_resolver.get_coreclr_library_version(), &_clr_library_version);
+            }
+
+            properties.add(common_property::TrustedPlatformAssemblies, _tpa_paths_cstr.data());
+            properties.add(common_property::NativeDllSearchDirectories, _native_dirs_cstr.data());
+            properties.add(common_property::PlatformResourceRoots, _resources_dirs_cstr.data());
+            properties.add(common_property::AppDomainCompatSwitch, "UseLatestBehaviorWhenTFMNotSpecified");
+            properties.add(common_property::AppContextBaseDirectory, _app_base_cstr.data());
+            properties.add(common_property::AppContextDepsFiles, _app_context_deps.data());
+            properties.add(common_property::FxDepsFile, _fx_deps.data());
+            properties.add(common_property::ProbingDirectories, _probe_directories.data());
+            properties.add(common_property::FxProductVersion, _clr_library_version.data());
+
+            if (!clrjit_path.empty())
+            {
+                pal::pal_clrstring(clrjit_path, &_clrjit_path_cstr);
+                properties.add(common_property::JitPath, _clrjit_path_cstr.data());
+            }
+
+            bool set_app_paths = false;
+
+            // Runtime options config properties.
+            for (int i = 0; i < _hostpolicy_init.cfg_keys.size(); ++i)
+            {
+                // Provide opt-in compatible behavior by using the switch to set APP_PATHS
+                if (pal::cstrcasecmp(_hostpolicy_init.cfg_keys[i].data(), "Microsoft.NETCore.DotNetHostPolicy.SetAppPaths") == 0)
+                {
+                    set_app_paths = (pal::cstrcasecmp(_hostpolicy_init.cfg_values[i].data(), "true") == 0);
+                }
+
+                properties.add(_hostpolicy_init.cfg_keys[i].data(), _hostpolicy_init.cfg_values[i].data());
```

Comment:
@vitek-karas Do you know why we are pushing this into the coreclr property bag? I am unable find any consumption of this property in the coreclr repo? Looks like it is purely for policy, so I don't think we should be propagating it. Unless there is some argument to be made for looking at why we set the `APP*PATHS` properties when debugging coreclr.

Reply:
I don't know the answer to this. I know that they are exposed through `AppContext.GetData` so there could be use cases in managed code. But I suspect it was basically a "why not" solution. Which is not a great reason. Unfortunately I don't think we can remove it - it would be a breaking change and we don't really have a good enough reason (other than it should not have been done in the first place)